app/controllers/katello/concerns/hosts_controller_extensions.rb

```diff
@@ -9,9 +9,9 @@ module HostsControllerExtensions
 
         def destroy
           sync_task(::Actions::Katello::Host::Destroy, @host)
-          process_success(:success_redirect => hosts_path)
+          process_success
         rescue StandardError => ex
-          process_error(:object => @host, :error_msg => ex.message)
+          process_error(:object => @host, :error_msg => ex.message, :redirect => saved_redirect_url_or(send("#{controller_name}_url")))
```

Comment:
Could you explain why this change is necessary?

Reply:
Basically, we store previous action against particular controller in session object.
`saved_redirect_url_or` method returns URL for given controller from session and if session don't have it returns default path. Here its hosts_path.
(https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L337)

If we don't pass `redirect` or `render`, in process_error it takes default path -> https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L307

Instead of overriding in process_error method as it can impact other things, I have passed it here.

Reply:
I see. So, is this a necessary change to fix the bug? It seems like it would work without it.

Reply:
@akofink ,
Yes, because I don't think that to modify it into `process_error` would be good idea if it is going to impact other requests.
In method `process_success`, it is already handled. 
( https://github.com/theforeman/foreman/blob/develop/app/controllers/application_controller.rb#L294)
Sorry but I did get how it would work without passing `redirect` key in argument hash?
Am I missing something here?

Reply:
I think I was the one missing something. :nerd_face: After talking with another member of the team, it seems perfectly reasonable.